mobile-blazor-bindings/walkthroughs/shell-navigation-manager.md

```diff
@@ -0,0 +1,95 @@
+
+For a sample of ShellNavigationManager see the [Xaminals sample in the Mobile Blazor Bindings repo](https://github.com/xamarin/MobileBlazorBindings/tree/master/samples/MobileBlazorBindingsXaminals).
+
+## Setup
+To use the ShellNavigationManager you'll need to have a Shell as the MainPage in your app.
+
+To start off here is a fairly simple Shell which will give us two tabs for a HomePage and an AboutPage. This should be in a file called `AppShell.razor`.
+
+```
+<Shell FlyoutBehavior="FlyoutBehavior.Disabled">
+    <TabBar>
+        <Tab>
+            <ShellContent>
+                <HomePage />
+            </ShellContent>
+
+            <ShellContent>
+                <AboutPage />
+            </ShellContent>
+        </Tab>
+    </TabBar>
+</Shell>
+```
+
+*Note*: The HomePage and AboutPage must have *ContentPage* as their root element ot be used in Shell.
+
+
+
+Inside App.cs, your AppShell needs to be set as the MainPage of your app. This is done by calling AddComponent. Setting the Shell as the MainPage is done in the background so we also have to set a blank ContentPage as a placeholder while it is loading.
+
+To enable ShellNavigationManager and make it available for our pages we add it as a singleton service.
+
+This should give you an App constructor something along the lines of:
+
+```
+public App()
+{
+    AppHost = MobileBlazorBindingsHost.CreateDefaultBuilder()
+        .ConfigureServices((hostContext, services) =>
+        {
+            // Register app-specific services
+            //services.AddSingleton<AppState>();
+            services.AddSingleton<ShellNavigationManager>();
+        })
+        .Build();
+
+    MainPage = new ContentPage();
+    AppHost.AddComponent<AppShell>(parent: this);
+}
+```
+
+## Registering Routes
+Route registration occurs in each razor component that you want to be able to navigate to using the `@page` directive followed by a string Uri. This Uri must start with a slash.
+
+For example here's a contact page with the route "/contact"
+
+```
+@page "/contact"
+<ContentView>
+    <StackLayout>
+        <Label Text="Phone Number:"></Label>
+        <Label Text="123456"></Label>
+    </StackLayout>
+</ContentView>
+```
+
+*Note*: At present target pages must have a *ContentView* as their root element. This differs from the *ContentPage* which must be used for children in a Shell.
+
+## Navigating to a page
+Navigating to a page is archived with an instance of the ShellNavigationManager which you can access in your components using the `@inject` directive. Once you have an instance you call NavigateToAsync() with the Uri for the page you want to open.
+
+For example in my AboutPage I want to navigate to the ContactPage.
+
+```
+@inject ShellNavigationManager NavigationManager
+<ContentPage>
+    <Stack
```

Comment:
Looks like the sample is cut off here? Was it also meant to show code to call into the NavigationManager and do the navigation?

Reply:
Ya, I just gave up half way through typing this sample. I've finished it now.